libkmod/libkmod-module.c

```diff
@@ -506,14 +506,14 @@ KMOD_EXPORT int kmod_module_get_filtered_blacklist(const struct kmod_ctx *ctx,
 	return kmod_module_apply_filter(ctx, KMOD_FILTER_BLACKLIST, input, output);
 }
 
-static void module_get_dependencies_noref(const struct kmod_module *mod)
+static void module_get_dependencies_noref(struct kmod_module *mod)
```

Comment:
AFAICT the premise is that since `kmod_module` and friends are opaque from the end user POV, modifying internal state should be safe (kind of).

That said, internal API which knows the distinction shouldn't be (ab)using that IMHO.